### PR TITLE
Boost: Add jetpack compatibility tests

### PR DESCRIPTION
**Pull request by karthikax**
#### Changes proposed in this Pull Request:

* Add tests for Jetpack compatibility
* List according to #22019:
    * Core Test Id 9
    * Core Test Id 10
* To retrieve the core Jetpack module status, this PR also adds a function `isModuleActive` to e2e-commons - prerequisites.js

#### Jetpack product discussion

Slack : archives/C0232B8CUHG/p1639639809027900

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Run tests as described here: #21819
* Verify that all the tests pass including the ones added in this PR!

### CODE REVIEW

**Comment by eAdnan007** [comment]
The changes look good. But seems like jetpack CLI is missing and tests are failing as a result?

**Comment by karthikax** [comment]
Oh, Yes. Not sure how this was passing with my local setup. Thanks, @eAdnan007.
Fixing this now!

**Review by eAdnan007 — DISMISSED** [review]
👍🏼

**Comment by karthikax** [comment]
Thanks, @eAdnan007. I am requesting team review as well since PR includes changes in tools/e2e-commons.

File: projects/plugins/boost/tests/e2e/specs/jetpack-compatibility.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import { JetpackBoostPage } from '../lib/pages/index.js';
+import { prerequisitesBuilder, isModuleActive } from 'jetpack-e2e-commons/env/prerequisites.js';
+import { activateModules, deactivateModules } from 'jetpack-e2e-commons/env/index.js';
+import {
+	boostPrerequisitesBuilder,
+	activateModules as activateBoostModules,
+	deactivateModules as deactivateBoostModules,
+} from '../lib/env/prerequisites.js';
+
+let jetpackBoostPage;
```
Comment (brbrr) [inline projects/plugins/boost/tests/e2e/specs/jetpack-compatibility.test.js:11]: Is there a reason for this variable to be global?

**Review by brbrr — APPROVED** [review]
Looks good!

I left a small nit-pick, but feel free to ignore it :)

File: projects/plugins/boost/tests/e2e/specs/jetpack-compatibility.test.js [hunk 2]
```diff
@@ -31,8 +29,8 @@ test.describe( 'Jetpack compatibility', () => {
 		await prerequisitesBuilder( page ).withActiveModules( [ 'lazy-images' ] ).build();
 		await deactivateModules( [ 'lazy-images' ] );
 
-		jetpackBoostPage = await JetpackBoostPage.visit( page );
-		expect( await jetpackBoostPage.isModuleEnabled( 'lazy-images' ) ).toEqual( false );
+		const jetpackBoostPage = await JetpackBoostPage.visit( page );
+		await expect( await jetpackBoostPage.isModuleEnabled( 'lazy-images' ) ).toEqual( false );
```
Comment (brbrr) [inline projects/plugins/boost/tests/e2e/specs/jetpack-compatibility.test.js:33]:
```suggestion
		expect( await jetpackBoostPage.isModuleEnabled( 'lazy-images' ) ).toEqual( false );
```

File: projects/plugins/boost/tests/e2e/specs/jetpack-compatibility.test.js (hunk 1 above)
Comment (karthikax) [reply projects/plugins/boost/tests/e2e/specs/jetpack-compatibility.test.js]: Started with this to manipulate the same page with multiple tests. It is no more valid. Thanks for pointing out @brbrr

**Review by brbrr — APPROVED** [review]

**Pull request merged**